manager/integration/tests/test_ha.py

```diff
@@ -2742,7 +2997,41 @@ def test_engine_image_not_fully_deployed_perform_auto_upgrade_engine():
     6. In a 2-min retry, verify that Longhorn upgrades the engine image of
        vol-1 and vol-2.
     """
-    pass
+    prepare_engine_not_fully_deployed_evnironment(client, core_api)
+
+    volume1 = create_and_check_volume(client, "vol-1", num_of_replicas=2,
+                                      size=str(3 * Gi))
+
+    volume2 = create_and_check_volume(client, "vol-2", num_of_replicas=2,
+                                      size=str(3 * Gi))
+
+    engine_upgrade_image, new_img = \
+        prepare_upgrade_image_not_fully_deployed_evnironment(client)
+
+    volume1.engineUpgrade(image=engine_upgrade_image)
+    volume2.engineUpgrade(image=engine_upgrade_image)
+    volume1 = wait_for_volume_current_image(client, volume1.name,
+                                            engine_upgrade_image)
+    volume2 = wait_for_volume_current_image(client, volume2.name,
+                                            engine_upgrade_image)
+
+    default_img = common.get_default_engine_image(client)
+    wait_for_engine_image_ref_count(client, default_img.name, 0)
+
+    volume2.attach(hostId=get_self_host_id())
```

Comment:
> 4. Attach vol-2 to node node-2

nit: `node-2` looks inconsistent with the implementation, suggest updating the test step.